sc-web/components/scgg/src/scgg-struct.js

```diff
@@ -256,112 +250,250 @@ function scggScStructTranslator(_editor, _sandbox) {
                 var dfd = new jQuery.Deferred();
                 
-                // create sc-struct nodes
-                var scAddrGen = function(c) {
-                    var dfd = new jQuery.Deferred();
-                    
-                    if (c.sc_addr)
-                        dfd.resolve();
-                    else {
-                        window.sctpClient.create_node(sc_type_const | sc_type_node | sc_type_node_struct).done(function (node) {
-                            c.setScAddr(node);
-                            c.setObjectState(SCggObjectState.NewInMemory);
-                            objects.push(c);
-                            if (c.text) {
-                                translateGtIdentifier(c)
-                                    .done(dfd.resolve)
-                                    .fail(dfd.reject);
-                            } else {
-                                dfd.resolve();
+                var implFunc = function(node){
+                    var dfdNode = new jQuery.Deferred();
+                        window.sctpClient.create_node(sc_type_node | sc_type_const).done(function (nodeAddr) {
+
+                            node.setScAddr(nodeAddr);
+                            window.sctpClient.create_arc(sc_type_arc_pos_const_perm, addrStruct, nodeAddr).done(function (arcSystemIdentifier){
+                                window.sctpClient.create_arc(sc_type_arc_pos_const_perm, SCggKeynodesHandler.scKeynodes.rrel_vertex, arcSystemIdentifier)
+                                .done(dfdNode.resolve)
+                                .fail(dfdNode.reject);
+                                //add idtf changed 
+                            if ((node.text !== '') && (node.text !== null) && (node.text !== undefined)) {
+                                translateGtIdentifier(node)
+                                .done(dfdNode.resolve)
+                                .fail(dfdNode.reject);
                             }
-                        });
-                    }
+                            else {
+                                dfdNode.resolve();
 
-                    return dfd.promise();
-                };
-                var funcs = [];
-                for (var i = 0; i < editor.scene.contours.length; ++i){
-                    editor.scene.contours[i].addNodesWhichAreInContourPolygon(editor.scene.nodes);
-                    editor.scene.contours[i].addNodesWhichAreInContourPolygon(editor.scene.links);
-                    editor.scene.contours[i].addEdgesWhichAreInContourPolygon(editor.scene.edges);
-                    funcs.push(fQueue.Func(scAddrGen, [ editor.scene.contours[i] ]));
+                            }
+                            
+                            })
+                            //leave it for me right now :D
+                            .fail(function() {
+                                console.log("WTF THIS MISTAKE " + addrStruct + " " + nodeAddr )
```

Comment:
can not create arc between addrStruct and nodeAddr

SCg do it yourself!!! check